### PR TITLE
feat: Add securityContext support at container level in helm chart templates

### LINKED ISSUE #3685: Add securityContext support at container level in helm chart templates
### What is the problem you're trying to solve

In the chart template files, the admission.yaml manifest does not support the "securityContext" section at container level, both for the deployment and job definition. 

### Describe the solution you'd like

Add following to admission.yaml:
```
...
---
apiVersion: apps/v1
kind: Deployment
...
spec:
...
  template:
  ...
    spec:
    ...
      containers:
          ...
          securityContext:
            {{- toYaml .Values.custom.default_csc | nindent 12 }}
          ...
...
---
apiVersion: batch/v1
kind: Job
...
spec:
...
  template:
    spec:
    ...
      containers:
          ...
          securityContext:
            {{- toYaml .Values.custom.default_csc | nindent 12 }}
          ...
...
```

This way, one could define the "default_csc" section in helm's values.yaml file to customize securityContext, in order to match its k8s cluster security constraints.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lekaf974**
Fix #3685

### CODE REVIEW

**Review by hwdef — COMMENTED** [review]
Should securityContext be added to scheduler and controller, and support setting different securityConetxt for admission, scheduler and controller?

**Comment by Monokaix** [comment]
> Should securityContext be added to scheduler and controller, and support setting different securityConetxt for admission, scheduler and controller?

+1
The original issue mentioned that controller and scheduler also need csc: )

**Comment by lekaf974** [comment]
Perfect will add changes for the scheduler

**Comment by lekaf974** [comment]
should I split csc one for admission and one for scheduler ?
For example:
- admission_default_csc
- scheduler_default_csc

**Comment by googs1025** [comment]
ci may fail. We are trying to fix it in this PR. Sorry for the inconvenience. :)

File: installer/helm/chart/volcano/values.yaml [hunk 1]
```diff
@@ -147,3 +147,19 @@ custom:
...
+
+# Specify container security context for admission
+# For example:
+#
+# admission_default_csc:
+#   allowPrivilegeEscalation: false
+#   runAsUser: 2000
+  admission_default_csc: ~
+
+# Specify container security context for scheduler
```
Comment (Monokaix) [inline installer/helm/chart/volcano/values.yaml:159]: We can put these `admission_default_csc: ~` and `scheduler_default_csc: ~` together.

**Comment by Monokaix** [comment]
controller is also needed: )

File: installer/helm/chart/volcano/values.yaml (hunk 1 above)
Comment (lekaf974) [reply installer/helm/chart/volcano/values.yaml:159]: You mean using the same definition for both ?

Comment (lekaf974) [reply installer/helm/chart/volcano/values.yaml:159]: forgot about it, I saw how other fields were made so I'll follow the same behavior

Comment (lekaf974) [reply installer/helm/chart/volcano/values.yaml]: Done

Comment (Monokaix) [reply installer/helm/chart/volcano/values.yaml]: Ok, and please also squash it to one commit.

**Comment by lekaf974** [comment]
> ci may fail. We are trying to fix it in this PR. Sorry for the inconvenience. :)

something similar to #3708

**Comment by lekaf974** [comment]
> controller is also needed: )

Done

**Comment by Monokaix** [comment]
Please also squash it to one commit: )

**Comment by lekaf974** [comment]
> Please also squash it to one commit: )

Done

**Review by hwdef — COMMENTED** [review]
/lgtm

**Pull request merged**